cli/ttl.go

```diff
@@ -91,37 +91,30 @@ func ttlset() *cobra.Command {
 				_, _ = fmt.Fprintf(cmd.OutOrStdout(), "warning: ttl rounded down to %s\n", truncated)
 			}
 
-			if changed, newDeadline := changedNewDeadline(workspace, truncated); changed {
-				// For the purposes of the user, "less than a minute" is essentially the same as "immediately".
-				timeRemaining := time.Until(newDeadline).Truncate(time.Minute)
-				humanRemaining := "in " + timeRemaining.String()
-				if timeRemaining <= 0 {
-					humanRemaining = "immediately"
-				}
-				_, err = cliui.Prompt(cmd, cliui.PromptOptions{
-					Text: fmt.Sprintf(
-						"Workspace %q will be stopped %s. Are you sure?",
-						workspace.Name,
-						humanRemaining,
-					),
-					Default:   "yes",
-					IsConfirm: true,
-				})
-				if err != nil {
-					if errors.Is(err, cliui.Canceled) {
-						return nil
-					}
-					return err
-				}
-			}
-
 			millis := truncated.Milliseconds()
 			if err = client.UpdateWorkspaceTTL(cmd.Context(), workspace.ID, codersdk.UpdateWorkspaceTTLRequest{
 				TTLMillis: &millis,
 			}); err != nil {
 				return xerrors.Errorf("update workspace ttl: %w", err)
 			}
 
+			if ptr.NilOrEmpty(workspace.AutostartSchedule) {
+				_, _ = fmt.Fprintf(cmd.OutOrStdout(), "%q will shut down %s after start.\n", workspace.Name, truncated)
+				return nil
+			}
+
+			sched, err := schedule.Weekly(*workspace.AutostartSchedule)
+			if err != nil {
+				return xerrors.Errorf("parse workspace schedule: %w", err)
+			}
+
+			nextShutdown := sched.Next(time.Now()).Add(truncated).In(sched.Location())
```

Comment:
Thought: If the workspace is currently running, should we mention that it's TTL is unaffected?

I haven't tried this out yet, but I imagine that if I ran this command and the workspace was running, seeing (only) that it will shut down tomorrow at X would be a bit confusing (I'm assuming this would, maybe I'm wrong).

Reply:
Added an extra note.